onnxruntime/core/providers/cpu/tensor/copy.h

```diff
@@ -1,27 +1,306 @@
-#include "core/providers/common.h"
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
 #include "core/platform/threadpool.h"
+#include "core/providers/common.h"
+#include "core/providers/op_kernel_type_control.h"
+#include "core/providers/op_kernel_type_control_utils.h"
+
 #include <vector>
 
 namespace onnxruntime {
 
 void CoalesceDimensions(
     std::initializer_list<std::reference_wrapper<std::vector<int64_t>>>&& tensors_strides, std::vector<int64_t>& shape);
 
+std::vector<int64_t> StridesForTensor(const Tensor& tensor);
+
+namespace {
```

Comment:
maybe use a name like "copy_detail" instead as it's in the header file so we don't want the internal linkage from the unnamed namespace.